pytorch_lightning/overrides/distributed.py

```diff
@@ -75,3 +76,70 @@ def prepare_for_backward(model: DistributedDataParallel, output: Any):
             model.reducer.prepare_for_backward([])
     else:
         model.require_forward_param_sync = False
+
+
+class UnrepeatedDistributedSampler(DistributedSampler):
+    """
+    A fork of the pytorch DistributedSampler that doesn't repeat data, instead
+    allowing the number of batches per process to be off-by-one from each other.
+    This makes this sampler usable for validation (it's deterministic and
+    doesn't require shuffling). It is potentially unsafe to use this sampler for
+    training, because during training the DistributedDataParallel syncs buffers
+    on each forward pass, so it could freeze if one of the processes runs one
+    fewer batch. During validation, buffers are only synced on the first batch,
```

Comment:
We only use for predicting but the comment says it can also be for validation?

Reply:
Good catch !